File: src/main/java/komu/blunt/eval/TypeCheckException.java
package komu.blunt.eval;

import komu.blunt.analyzer.AnalyzationException;
import komu.blunt.types.checker.UnificationException;

public class TypeCheckException extends AnalyzationException {

    public TypeCheckException(String message) {
        super(message);
    }

    public TypeCheckException(UnificationException e) {
        super(e.getMessage());
    }

}

File: src/main/kotlin/komu/blunt/eval/TypeCheckException.kt
package komu.blunt.eval

import komu.blunt.analyzer.AnalyzationException
import komu.blunt.types.checker.UnificationException

class TypeCheckException(message: String) : AnalyzationException(message) {
    this(e: UnificationException): this(e.getMessage().sure()) { }
}
